src/worker/jobs.ts

```diff
@@ -153,6 +118,118 @@ export default async (): Promise<void> => {
     });
 };
 
+function reconnectChainSubscriber() {
+    waitingForResponseAfterCrash = true;
+    subscribers.stop();
+    // if a second crash happen before recovering from the first
+    // it will get here again. Clear past time interval
+    // and start again.
+    if (intervalWhenCrash !== undefined) {
+        clearInterval(intervalWhenCrash);
+    }
+    intervalWhenCrash = setInterval(() => {
+        if (usingFallbackUrl) {
+            providerFallback
+                .getBlockNumber()
+                .then(() => {
+                    successfullAnswersAfterCrash += 1;
+                    // require 5 successfull answers, to prevent two or more crashes in row
+                    if (successfullAnswersAfterCrash < 5) {
+                        Logger.error(
+                            'Got ' +
+                                successfullAnswersAfterCrash +
+                                '/5 sucessfull responses form json rpc provider (fallback)...'
+                        );
+                    } else {
+                        Logger.error(
+                            'Reconnecting json rpc provider (fallback)...'
+                        );
+                        subscribers.recover();
+                        clearInterval(intervalWhenCrash!);
+                        intervalWhenCrash = undefined;
+                        waitingForResponseAfterCrash = false;
+
+                        // After 30 min, try to connect with the principal provider again
+                        schedule.scheduleJob(Date.now() + 60000, () => {
+                            Logger.info(
+                                'Conecting with the principal provider'
+                            );
+                            subscribers.stop();
+                            subscribers = startChainSubscriber();
+                            usingFallbackUrl = false;
+                            failedAnswers = 0;
+                            successfullAnswersAfterCrash = 0;
+                            reconnectChainSubscriber();
+                        });
```

Comment:
if it fails, does it retry again in 30 mins?

Reply:
yes, after 30 minutes the flow restarts. Trying to connect to the principal RPC, and in case of continuing error, maintain connected with fallback (30 minutes later retry)

Reply:
ahh correct, because it actually calls the method again.